### PR TITLE
src/components: add final screen for password reset

### PR DESCRIPTION
**Pull request by maceksimon**
* Add state `reset-finished` to `FormLogin.vue` showing confirmation after password reset
* Add translations
* Add tests

### CODE REVIEW

File: src/components/FormLogin.vue [hunk 1]
```diff
@@ -238,10 +253,67 @@ export default defineComponent({
         type="submit"
         color="primary"
         :label="$t('login.form.submitPasswordReset')"
+        @click="formState = 'reset-finished'"
```
Comment (tmszi) [inline src/components/FormLogin.vue:256]: We need validate email form field value first.

**Comment by tmszi** [comment]
Another bug is (check the web browser dev Console): 

```
04:33:58.061 [intlify] Not found 'login.bannerAppButtons.googlePlayAltText' key in 'en' locale messages. [shared.mjs:159:16](http://localhost:9000/node_modules/@intlify/shared/dist/shared.mjs)
04:33:58.061 [intlify] Not found 'login.bannerAppButtons.appStoreAltText' key in 'en' locale messages. [shared.mjs:159:16](http://localhost:9000/node_modules/@intlify/shared/dist/shared.mjs)
```

File: src/components/FormLogin.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/FormLogin.vue]: Added email validation to both inputs and validation tests in `FormLogin.cy.js`.

File: src/components/FormLogin.vue [hunk 2]
```diff
@@ -57,12 +74,25 @@ export default defineComponent({
       // noop
     };
 
+    const onClickFormPasswordResetBtn = (): void => {
+      if (!isValid(formPasswordReset.email)) {
```
Comment (tmszi) [inline src/components/FormLogin.vue:78]:
```ts
if (!isValid(formPasswordReset.email) || !isEmail(formPasswordReset.email)) return;
```

File: src/components/__tests__/FormLogin.cy.js [hunk 3]
```diff
@@ -173,6 +174,99 @@ describe('<FormLogin>', () => {
...
+        .and('have.text', i18n.global.t('login.form.submitNewPassword'));
+    });
+
+    it('validates login form user inputs', () => {
+      cy.dataCy('form-login-submit-login').should('be.visible').click();
+      // validate email required
+      cy.dataCy('form-login-email')
+        .find('.q-field__messages')
+        .should('be.visible')
+        .should('contain', i18n.global.t('login.form.messageEmailReqired'));
```
Comment (tmszi) [inline src/components/__tests__/FormLogin.cy.js:235]: `.and()`

File: src/components/__tests__/FormLogin.cy.js [hunk 4]
```diff
@@ -173,6 +174,99 @@ describe('<FormLogin>', () => {
...
+      // validate email format
+      cy.dataCy('form-login-email-input').type('qw123@qw');
+      cy.dataCy('form-login-email')
+        .find('.q-field__messages')
+        .should('be.visible')
+        .should('contain', i18n.global.t('login.form.messageEmailInvalid'));
```
Comment (tmszi) [inline src/components/__tests__/FormLogin.cy.js:241]: `.and()`

File: src/components/__tests__/FormLogin.cy.js [hunk 5]
```diff
@@ -173,6 +174,99 @@ describe('<FormLogin>', () => {
...
+      // validate password required
+      // first make email pass
+      cy.dataCy('form-login-email-input').type('.com');
+      cy.dataCy('form-login-submit-login').should('be.visible').click();
+      cy.dataCy('form-login-password')
+        .find('.q-field__messages')
+        .should('be.visible')
+        .should('contain', i18n.global.t('login.form.messagePasswordRequired'));
```
Comment (tmszi) [inline src/components/__tests__/FormLogin.cy.js:249]: `.and()`

File: src/components/__tests__/FormLogin.cy.js [hunk 6]
```diff
@@ -173,6 +174,99 @@ describe('<FormLogin>', () => {
...
+    });
+
+    it('validates password reset form user inputs', () => {
+      cy.dataCy('form-login-forgotten-password').should('be.visible').click();
+      cy.dataCy('form-password-reset-submit').should('be.visible').click();
+      // validate email required
+      cy.dataCy('form-password-reset-email')
+        .find('.q-field__messages')
+        .should('be.visible')
+        .should(
```
Comment (tmszi) [inline src/components/__tests__/FormLogin.cy.js:259]: `.and()`

File: src/components/__tests__/FormLogin.cy.js [hunk 7]
```diff
@@ -173,6 +174,99 @@ describe('<FormLogin>', () => {
...
+          'contain',
+          i18n.global.t('login.form.messagePasswordResetReqired'),
+        );
+      // validate email format
+      cy.dataCy('form-password-reset-email-input').type('qw123@qw');
+      cy.dataCy('form-password-reset-email')
+        .find('.q-field__messages')
+        .should('be.visible')
+        .should('contain', i18n.global.t('login.form.messageEmailInvalid'));
```
Comment (tmszi) [inline src/components/__tests__/FormLogin.cy.js:268]: `.and()`

File: src/i18n/en.toml [hunk 8]
```diff
@@ -145,17 +145,22 @@ deployedAppVersion = "Deployed app version"
 [login.form]
 titleLogin = "Login"
 titlePasswordReset = "Forgotten password"
+titleResetFinished = "You will receive a password reset link in your email"
 descriptionPasswordReset = "We will email you a link to reset your password."
+descriptionResetFinished = "We have just emailed you a link to reset your password at anonym@dopracenakole.cz. If you can't find the link, check your spam folder."
```
Comment (tmszi) [inline src/i18n/en.toml:150]: Perhaps this email address anonym@dopracenakole.cz should have a place in the global configuration file.

File: src/components/FormLogin.vue [hunk 9]
```diff
@@ -45,10 +53,19 @@ export default defineComponent({
...
+    const formState = ref<'login' | 'password-reset' | 'reset-finished'>(
+      'login',
+    );
+
+    const backgroundColor = rideToWorkByBikeConfig.colorWhiteOpacity;
 
     const isValid = (val: string): boolean => val?.length > 0;
 
+    const isEmail = (value: string): boolean => {
+      const regex = /^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+\.[a-zA-z]{2,3}$/;
```
Comment (tmszi) [inline src/components/FormLogin.vue:65]: [source](https://en.wikipedia.org/wiki/Email_address): _dot ., provided that it is not the first or last character and provided also that it does not appear consecutively (e.g., John..Doe@example.com is not allowed)_

File: src/components/FormLogin.vue (hunk 2 above)
Comment (tmszi) [reply src/components/FormLogin.vue:78]: Better

```ts
if (isValid(formPasswordReset.email) && isEmail(formPasswordReset.email)) formState.value = 'reset-finished';
```

Comment (maceksimon) [reply src/components/FormLogin.vue]: Edited

File: src/components/__tests__/FormLogin.cy.js (hunk 4 above)
Comment (maceksimon) [reply src/components/__tests__/FormLogin.cy.js]: Fixed.

File: src/components/__tests__/FormLogin.cy.js (hunk 3 above)
Comment (maceksimon) [reply src/components/__tests__/FormLogin.cy.js]: Fixed.

File: src/components/__tests__/FormLogin.cy.js (hunk 5 above)
Comment (maceksimon) [reply src/components/__tests__/FormLogin.cy.js]: Fixed.

File: src/components/__tests__/FormLogin.cy.js (hunk 6 above)
Comment (maceksimon) [reply src/components/__tests__/FormLogin.cy.js]: Fixed.

File: src/components/__tests__/FormLogin.cy.js (hunk 7 above)
Comment (maceksimon) [reply src/components/__tests__/FormLogin.cy.js]: Fixed.

File: src/components/FormLogin.vue (hunk 9 above)
Comment (maceksimon) [reply src/components/FormLogin.vue]: Edited based on linked requirements ([regex source](https://uibakery.io/regex-library/email)). Added email validation test.
(As discussed, certain valid but very unusual addresses are currently not supported in favour of a more concise regex rule.)

File: src/i18n/en.toml (hunk 8 above)
Comment (maceksimon) [reply src/i18n/en.toml]: Good point. Edited.

**Review by tmszi — APPROVED** [review]

**Pull request merged**